data/src/test/java/com/microsoft/azure/kusto/data/KustoResultMapperTest.java

```diff
@@ -0,0 +1,246 @@
+	}
+	
+	public static class TestPojo {
+		int id;
+		String name;
+		ZonedDateTime zonedTime;
+		Instant instantTime;
+		TestEnum testEnum;
+		UUID uuid;
+		
+		public void setId(int id) {
+			this.id = id;
+		}
+		
+		public void setName(String name) {
+			this.name = name;
+		}
+		
+		public void setZonedTime(ZonedDateTime time) {
+			this.zonedTime = time;
+		}
+		
+		public void setInstantTime(Instant time) {
+			this.instantTime = time;
+		}
+		
+		public void setUuid(UUID uuid) {
+			this.uuid = uuid;
+		}
+		
+		public void setTestEnum(TestEnum testEnum) {
+			this.testEnum = testEnum;
+		}
+	}
+	
+	static final KustoResultMapper<TestPojo> namedMapper = KustoResultMapper.newBuilder(TestPojo::new)
+			.addNonNullableColumn(KustoType.INTEGER, "Id", TestPojo::setId).addNullableColumn(KustoType.STRING, "Name", TestPojo::setName)
+			.addNullableColumn(KustoType.DATETIME_ZONED_DATE_TIME, "TimeColumn", TestPojo::setZonedTime)
+			.addNullableColumn(KustoType.GUID_UUID, "Guid", TestPojo::setUuid)
+			.addNullableColumn(KustoType.STRING, "Enum", (r, t) -> r.setTestEnum(t != null ? TestEnum.valueOf(t) : null))
+			.addNullableColumn(KustoType.DATETIME_INSTANT, "Instant", TestPojo::setInstantTime).build();
+	
+	static final KustoResultMapper<TestPojo> ordinalMapper = KustoResultMapper.newBuilder(TestPojo::new)
+			.addNonNullableColumn(KustoType.INTEGER, 1, TestPojo::setId).addNullableColumn(KustoType.STRING, 2, TestPojo::setName)
+			.addNullableColumn(KustoType.DATETIME_ZONED_DATE_TIME, 3, TestPojo::setZonedTime).addNullableColumn(KustoType.GUID_UUID, 4, TestPojo::setUuid)
+			.addNullableColumn(KustoType.STRING, "Enum", (r, t) -> r.setTestEnum(t != null ? TestEnum.valueOf(t) : null))
+			.addNullableColumn(KustoType.DATETIME_INSTANT, 6, TestPojo::setInstantTime).build();
+	
+	static final KustoResultMapper<TestPojo> mixedMapper = KustoResultMapper.newBuilder(TestPojo::new)
+			.addNonNullableColumn(KustoType.INTEGER, "Id", 1, TestPojo::setId).addNullableColumn(KustoType.STRING, "Name", 2, TestPojo::setName)
+			.addNullableColumn(KustoType.DATETIME_ZONED_DATE_TIME, "TimeColumn", 3, TestPojo::setZonedTime)
+			.addNullableColumn(KustoType.GUID_UUID, "Guid", 4, TestPojo::setUuid)
+			.addNullableColumn(KustoType.STRING, "Enum", 5, (r, t) -> r.setTestEnum(t != null ? TestEnum.valueOf(t) : null))
+			.addNullableColumn(KustoType.DATETIME_INSTANT, "Instant", 6, TestPojo::setInstantTime).build();
+	
+	KustoResultSetTable resultSet = mock(KustoResultSetTable.class);
+	
+	@BeforeEach
+	public void prepareResultSet() {
+		when(this.resultSet.findColumn("Id")).thenReturn(1);
+		when(this.resultSet.findColumn("Name")).thenReturn(2);
+		when(this.resultSet.findColumn("TimeColumn")).thenReturn(3);
+		when(this.resultSet.findColumn("Guid")).thenReturn(4);
+		when(this.resultSet.findColumn("Enum")).thenReturn(5);
+		when(this.resultSet.findColumn("Instant")).thenReturn(6);
+	}
+	
+	void testNonNull(KustoResultMapper<TestPojo> mapper) throws Exception {
```

Comment:
testSingleNonNullable

Reply:
renamed all tests that test "extractSingle" - added "single" to all of them.